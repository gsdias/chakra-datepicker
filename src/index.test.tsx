import React from 'react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import { Simulate } from 'react-dom/test-utils'
import { ChakraProvider } from '@chakra-ui/react'
import Calendario from '.'

let container: any

beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
})

describe('Calendario', () => {
  it('Should render correctly', () => {
    const tree = renderer
      .create(
        <ChakraProvider>
          <Calendario />
        </ChakraProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Calls onclick action', () => {
    // const onClick = jest.fn()
    ReactDOM.render(
      <ChakraProvider>
        <Calendario />
      </ChakraProvider>,
      container,
    )
    const input = container.querySelector('input')
    Simulate.focus(input)
    const datePicker = container.querySelector('.datePicker')
    expect(datePicker).not.toBeNull()
    // expect(onClick).toHaveBeenCalledTimes(1)
  })
})
