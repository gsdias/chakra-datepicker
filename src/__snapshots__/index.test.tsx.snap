// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendario Should render correctly 1`] = `
Array [
  <div
    className="chakra-container datePicker css-1bqfmys"
  >
    <input
      className="chakra-input datePicker__input css-1c6j008"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      placeholder="Date"
      readOnly={false}
      value=""
    />
  </div>,
  <span />,
]
`;
